crates/bevy_reflect/bevy_reflect_derive/src/utility.rs

```diff
@@ -96,3 +98,42 @@ impl<T> ResultSifter<T> {
         }
     }
 }
+
+/// Converts an iterator over ignore behaviour of members to a bitset of ignored members.
+///
+/// Takes into account the fact that always ignored (non-reflected) members do not count as members.
+///
+/// # Example
+/// ```rust,ignore
+/// pub struct HelloWorld {
+///     reflected_field: u32      // index: 0
+///
+///     #[reflect(ignore)]
+///     non_reflected_field: u32  // index: N/A (not 1!)
+///
+///     #[reflect(skip_serializing)]
+///     non_serialized_field: u32 // index: 1
+/// }
+/// ```
+/// Would convert to the `0b01` bitset (i.e second field is NOT serialized)
+///
+pub(crate) fn members_to_serialization_blacklist<T>(member_iter: T) -> BitSet<u32>
+where
+    T: Iterator<Item = ReflectIgnoreBehaviour>,
+{
+    let mut bitset = BitSet::default();
+
+    member_iter.fold(0, |next_idx, member| {
+        if member == ReflectIgnoreBehaviour::IgnoreAlways {
+            bitset.insert(next_idx);
+            next_idx
+        } else if member == ReflectIgnoreBehaviour::IgnoreSerialization {
+            bitset.insert(next_idx);
+            next_idx + 1
+        } else {
+            next_idx + 1
+        }
```

Comment:
Nit: Could be a match
```suggestion
        match member {
        ReflectIgnoreBehaviour::IgnoreAlways | ReflectIgnoreBehaviour::IgnoreSerialization => {
            bitset.insert(next_idx);
            next_idx
        }
        _ => next_idx + 1
```

Reply:
I can change it to a match however do note that the all branches here are unique, note line 129 vs line 132

Reply:
Oops my mistake, yeah they'd all be unique